scalding-date/src/main/scala/com/twitter/scalding/DateOps.scala

```diff
@@ -33,26 +33,38 @@ object DateOps extends java.io.Serializable {
   val DATETIME_HMS_WITH_DASH = "yyyy-MM-dd HH:mm:ss"
   val DATETIME_HMSM_WITH_DASH = "yyyy-MM-dd HH:mm:ss.SSS"
 
-  private val DATE_RE = """\d{4}-\d{2}-\d{2}"""
-  private val SEP_RE = """(T?|\s*)"""
-  private val DATE_FORMAT_VALIDATORS = List(DATE_WITH_DASH -> new Regex("""^\s*""" + DATE_RE + """\s*$"""),
-    DATEHOUR_WITH_DASH -> new Regex("""^\s*""" + DATE_RE +
-      SEP_RE + """\d\d\s*$"""),
-    DATETIME_WITH_DASH -> new Regex("""^\s*""" + DATE_RE +
-      SEP_RE + """\d\d:\d\d\s*$"""),
-    DATETIME_HMS_WITH_DASH -> new Regex("""^\s*""" + DATE_RE +
-      SEP_RE + """\d\d:\d\d:\d\d\s*$"""),
-    DATETIME_HMSM_WITH_DASH -> new Regex("""^\s*""" + DATE_RE +
-      SEP_RE + """\d\d:\d\d:\d\d\.\d{1,3}\s*$"""))
+  sealed abstract class Format(val pattern: String, val validator: Regex)
+
+  object Format {
+    private val DATE_RE = """\d{4}-\d{2}-\d{2}"""
+    private val SEP_RE = """(T?|\s*)"""
+
```

Comment:
while we're at it, can you add `private val emptyBegin = """^\s*"""` and use it below. Also, can you refer to the `DateOps.DATETIME_...` strings instead of repeating them below?